crawler/helpers.py

```diff
@@ -144,52 +180,199 @@ def parse_csv(centre: Dict[str, str]) -> Tuple[str, List[str], List[Dict[str, st
         csvreader = DictReader(csvfile)
 
-        errors = check_for_required_fields(csvreader, centre)
-
-        errors, documents = add_extra_fields(csvreader, centre, errors)
+        check_for_required_fields(csvreader, centre)
+        errors, documents = add_extra_fields(csvreader, centre)
 
     return latest_file_name, errors, documents
 
 
-def get_latest_csv(centre_details: Dict[str, str]) -> str:
-    """Get the latest CSV file for the centre which matches the regex described in the centre
-    details file.
+def get_download_dir(centre: Dict[str, str]) -> str:
+    """Get the download directory where the files from the SFTP are stored.
 
     Arguments:
-        centre_details {Dict[str, str]} -- details of the centres
+        centre {Dict[str, str]} -- the centre in question
+
+    Returns:
+        str -- the download directory
+    """
+    download_dir = f"{DIR_DOWNLOADED_DATA}{centre['prefix']}/"
+
+    return download_dir
+
+
+def get_files_in_download_dir(centre: Dict[str, str], regex_field: str) -> List[str]:
+    """Get all the files in the download directory for this centre and filter the file names using
+    the regex described in the centre's 'regex_field'.
+
+    Arguments:
+        centre {Dict[str, str]} -- the centre in question
+        regex_field {str} -- the field name where the regex is found to filter the files by
+
+    Returns:
+        List[str] -- all the file names in the download directory after filtering
+    """
+    # get a list of files in the download directory
+    # https://stackoverflow.com/a/3207973
+    (_, _, files) = next(os.walk(PROJECT_ROOT.joinpath(get_download_dir(centre))))
+
+    pattern = re.compile(centre[regex_field])
+
+    # filter the list of files to only those which match the pattern
+    centre_files = list(filter(pattern.match, files))
+
+    return centre_files
+
+
+def get_latest_csv(centre: Dict[str, str], regex_field: str) -> str:
+    """Get the latest CSV file name for the centre which matches the regex described by the
+    'regex_field' for the centre.
+
+    Arguments:
+        centre {Dict[str, str]} -- the centre in question
 
     Returns:
         str -- file name of the CSV file to parse
     """
     logger.debug(
-        f"Getting latest CSV file for {centre_details['name']} using "
-        f"pattern {centre_details['sftp_file_regex']}"
+        f"Getting latest CSV file for {centre['name']} using " f"pattern {centre[regex_field]}"
     )
-    root = pathlib.Path(__file__).parent.parent
 
-    # https://stackoverflow.com/a/3207973
-    (_, _, files) = next(os.walk(root.joinpath(DIR_DOWNLOADED_DATA)))
-
-    pattern = re.compile(centre_details["sftp_file_regex"])
-    centre_files = filter(pattern.match, files)
+    centre_files = get_files_in_download_dir(centre, regex_field)
 
     files_with_time = {}
     for filename in centre_files:
-        if match := pattern.match(filename):
+        if match := re.compile(centre[regex_field]).match(filename):
```

Comment:
How fast is regex compiling? Is it worth caching it outside of the loop?

Reply:
Never thought of that. I think it is negligible. Each regex in the loop is potentially different so we would need to compile it.